libs/pipelines/api_v2_paths.py

```diff
@@ -15,40 +17,60 @@ def suffix(self):
             return "json"
 
 
-def _region_key(level):
-    if level is AggregationLevel.COUNTY:
-        return "counties"
-    if level is AggregationLevel.STATE:
-        return "states"
+@dataclasses.dataclass
+class APIOutputPathBuilder:
+    root: pathlib.Path
+    level: AggregationLevel
+
+    @property
+    def region_key(self):
+        if self.level is AggregationLevel.COUNTY:
+            return "counties"
+        if self.level is AggregationLevel.STATE:
+            return "states"
 
+        raise ValueError("Unsupported aggregation level")
 
-class APIOutputPathBuilder:
-    def __init__(self, root: pathlib.Path):
-        self.root = root
+    @property
+    def region_subdir(self):
+        if self.level is AggregationLevel.COUNTY:
+            return "county"
+        if self.level is AggregationLevel.STATE:
+            return "state"
+
+        raise ValueError("Unsupported aggregation level")
+
+    @property
+    def region_path(self) -> pathlib.Path:
```

Comment:
Based on the name I'd expect this to be a path of a file containing something about one region. Do you think region_subdir may be used outside this class? If not how about having it return path?